File: 2026/06-June/13-Saturday/solution.py
class Solution:
    def mapWordWeights(self, words: List[str], weights: List[int]) -> str:
        ans = []

        for word in words:
            total = sum(weights[ord(ch) - ord('a')] for ch in word)
            total %= 26
            ans.append(chr(ord('a') + (25 - total)))

        return "".join(ans)

File: 2026/06-June/13-Saturday/solution.cs
public class Solution
{
    public string MapWordWeights(string[] words, int[] weights)
    {
        StringBuilder ans = new StringBuilder();

        foreach (string word in words)
        {
            int sum = 0;

            foreach (char c in word)
            {
                sum += weights[c - 'a'];
            }

            sum %= 26;
            ans.Append((char)('a' + (25 - sum)));
        }

        return ans.ToString();
    }
}
